source/s3_meta_request.c

```diff
@@ -401,13 +405,7 @@ void aws_s3_meta_request_send_next_request(
         goto unlock;
     }
 
-    /* Try grabbing a request_desc from the retry queue. */
-    if (!aws_linked_list_empty(&meta_request->synced_data.retry_queue)) {
-        struct aws_linked_list_node *request_node = aws_linked_list_pop_front(&meta_request->synced_data.retry_queue);
-
-        request = AWS_CONTAINER_OF(request_node, struct aws_s3_request, node);
-        AWS_FATAL_ASSERT(request != NULL);
-    }
+    request = aws_s3_meta_request_retry_queue_pop_synced(meta_request);
```

Comment:
feels like this is simple enough now that unlock: could be dropped in place of a single if-else.

Reply:
But there would just be one right?

`
if (state == finished) {
   meta_request_already_finished = true
} else {
   request = aws_s3_meta_request_retry_queue_pop_synced(..);
}
`
 and the unlock label goes away

Reply:
You are correct.  I will fix.